app/models/solidus_paypal_braintree/gateway.rb

```diff
@@ -363,7 +374,7 @@ def paypal_payee_email_for(source, options)
     def customer_profile_params(payment)
       params = {}
 
-      if payment.source.try(:nonce)
+      if preferred_paypal_flow == 'vault' && payment.source.try(:nonce)
```

Comment:
Could we also use the `store_in_vault` method here?

Reply:
@tvdeyen I think this is needed even when you don't store the payment source.

Reply:
@tvdeyen sorry, I misread your initial comment. Will correct. :)